CMakeLists.txt

```diff
@@ -551,47 +507,82 @@ list(REMOVE_ITEM CMAKE_C_FLAGS_MINSIZEREL "-DNDEBUG")
+set(CMAKE_C_FLAGS_RELEASE "-O3" CACHE STRING "Release build flags" FORCE)
+# set the arch to use for Release build if provided
+if(ARCH)
+	MESSAGE("adding -march=${ARCH} to Release build")
+	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=${ARCH}" CACHE STRING "Release build flags" FORCE)
+	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -march=${ARCH}" CACHE STRING "Release build flags" FORCE)
+endif(ARCH)
+# add pentiumpro arch for windows builds with -O3 if no other arch provided, otherwise resulting executable may not work
+if(WIN32 AND NOT ARCH)
+	MESSAGE("WIN32 and no arch provided, defaulting to pentiumpro")
+	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=pentiumpro" CACHE STRING "Release build flags" FORCE)
+	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -march=pentiumpro" CACHE STRING "Release build flags" FORCE)
+endif(WIN32 AND NOT ARCH)
+
+# if compiling with LTO
+if("${ENABLE_LTO}" STREQUAL "yes")
+# clang and gcc require different parallelization options and linkers
+	if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+		MESSAGE("added -flto=thin to Release build")
+		set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -flto=thin" CACHE STRING "Release build flags with LTO" FORCE)
+		set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -flto=thin" CACHE STRING "Release build flags with LTO" FORCE)
+		
+		MESSAGE("Using Clang LLD linker")
+		set(CMAKE_EXE_LINKER_FLAGS_RELEASE "-s -fuse-ld=lld" CACHE STRING "Linker flag for building with LTO and clang" FORCE)
+	elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+		if(NOT LTO_JOBS)
+			MESSAGE("LTO_JOBS not set, defaulting to 1")
+			set(LTO_JOBS "1" CACHE STRING "Number of threads to use for LTO with gcc" FORCE)
+		endif(NOT LTO_JOBS)
+		
+		MESSAGE("added -flto=${LTO_JOBS} to Release build")
+		set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -flto=${LTO_JOBS}" CACHE STRING "Release build flags with LTO" FORCE)
+		set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -flto=${LTO_JOBS}" CACHE STRING "Release build flags with LTO" FORCE)
+		
+		MESSAGE("Using GCC gold linker")
+		set(CMAKE_EXE_LINKER_FLAGS_RELEASE "-s -fuse-ld=gold" CACHE STRING "Linker flag for building with LTO and gcc" FORCE)
+	endif()
 	
-	separate_arguments(CMAKE_C_FLAGS_OPTIMIZED)
-	list(REMOVE_ITEM CMAKE_C_FLAGS_OPTIMIZED "-march=native")
-	list(APPEND CMAKE_C_FLAGS_OPTIMIZED "-march=pentiumpro")
-	list(REMOVE_DUPLICATES CMAKE_C_FLAGS_OPTIMIZED)
-	string(REPLACE ";" " " CMAKE_C_FLAGS_OPTIMIZED
-		"${CMAKE_C_FLAGS_OPTIMIZED}")
-	set(CMAKE_C_FLAGS_OPTIMIZED "${CMAKE_C_FLAGS_OPTIMIZED}" CACHE STRING "added -march=pentiumpro" FORCE)
-	MESSAGE("added -march=pentiumpro to Optimized build")
-endif(WIN32)
+# set CMAKE_AR and CMAKE_RANLIB to use LTO-enabled variants if LTO is enabled
+	MESSAGE("Using gcc-ar and gcc-ranlib")
+	find_program(LTO_AR NAMES gcc-ar)
+	find_program(LTO_RANLIB NAMES gcc-ranlib)
+	set(CMAKE_AR "${LTO_AR}" CACHE STRING "Supports LTO" FORCE)
```

Comment:
Please remove unnecessary string substitution:

```
set(CMAKE_AR LTO_AR CACHE STRING "Supports LTO" FORCE)
```

Same for the other three lines.

Reply:
This does not work - making this change results in `make` trying to run:
`LTO_AR qc libwesnoth-game.a`
rather than:
`/usr/bin/gcc-ar qc libwesnoth-game.a`